instancio-core/src/main/java/org/instancio/internal/beanvalidation/HibernateBeanValidationProcessor.java

```diff
@@ -30,53 +29,46 @@
 
-final class HibernateBeanValidationProcessor extends AbstractBeanValidationProvider {
+import static org.instancio.internal.util.ExceptionHandler.runIgnoringTheNoClassDefFoundError;
 
-    private static final Set<Class<?>> PRIMARY = CollectionUtils.asSet(
-            EAN.class, CreditCardNumber.class, ISBN.class, LuhnCheck.class, URL.class, UUID.class);
+final class HibernateBeanValidationProcessor extends AbstractBeanValidationProvider {
 
     private final HibernateBeanValidationHandlerResolver resolver =
             HibernateBeanValidationHandlerResolver.getInstance();
 
-    @Override
-    public boolean isPrimary(final Class<? extends Annotation> annotationType) {
-        return PRIMARY.contains(annotationType);
+    HibernateBeanValidationProcessor() {
+        super(buildMap());
     }
 
-    @Override
-    protected Generator<?> resolveGenerator(
-            final Annotation annotation,
-            final GeneratorContext context) {
-
-        final Class<?> annotationType = annotation.annotationType();
-        if (annotationType == EAN.class) {
-            return getEanGenerator((EAN) annotation, context);
-        }
-        if (annotationType == LuhnCheck.class) {
-            return getLuhnGenerator((LuhnCheck) annotation, context);
-        }
-        if (annotationType == CreditCardNumber.class) {
-            return new CreditCardNumberGenerator(context);
-        }
-        if (annotationType == ISBN.class) {
-            return new IsbnGenerator(context);
-        }
-        if (annotationType == UUID.class) {
-            return new UUIDGenerator(context);
-        }
-        if (annotationType == URL.class) {
-            return getUrlGenerator((URL) annotation, context);
-        }
-
-        // should not be reachable if caller checked for supported annotations
-        throw new InstancioException("Unmapped primary annotation:  " + annotationType.getName());
+    private static Map<Class<? extends Annotation>, BiFunction<Annotation, GeneratorContext, Generator<?>>> buildMap() {
+        Map<Class<? extends Annotation>, BiFunction<Annotation, GeneratorContext, Generator<?>>> map = new HashMap<>();
+        runIgnoringTheNoClassDefFoundError(() ->
+                map.put(EAN.class, ((annotation, context) -> getEanGenerator((EAN) annotation, context)))
+        );
+        runIgnoringTheNoClassDefFoundError(() ->
+                map.put(LuhnCheck.class, ((annotation, context) -> getLuhnGenerator((LuhnCheck) annotation, context)))
+        );
+        runIgnoringTheNoClassDefFoundError(() ->
+                map.put(CreditCardNumber.class, ((annotation, context) -> new CreditCardNumberGenerator(context)))
+        );
+        runIgnoringTheNoClassDefFoundError(() ->
+                map.put(ISBN.class, ((annotation, context) -> new IsbnGenerator(context)))
+        );
+        runIgnoringTheNoClassDefFoundError(() ->
+                map.put(UUID.class, ((annotation, context) -> new UUIDGenerator(context)))
```

Comment:
Let's limit `runIgnoringTheNoClassDefFoundError` to annotations that are not present in all the versions (`UUID` is a good choice for this).

However, for annotations that are present in all (or most) versions, `runIgnoringTheNoClassDefFoundError`  should not be necessary.

Reply:
I still think that the `runIgnoringTheNoClassDefFoundError` method should be used for all annotations , because:
1) Using of `runIgnoringTheNoClassDefFoundError` method provides universal approach for registration of generators for any constraint annotation. By use of this approach, a developer can write valid code without thinking about how often the current annotation is used: is it present in all versions or not. I think it can simplify future support for `Instancio` tool.
2) It seems that it will be difficult to remember that some of annotations were added not in the first version of validation library. For example:
![Screenshot_20230303_091735](https://user-images.githubusercontent.com/3287489/222656977-559c97ae-5266-42e0-87e4-58b4ea628c6b.png)
Therefore if it would necessary to add the support some of these annotations, developer of `Instancio` tool should not forget to use `runIgnoringTheNoClassDefFoundError` method for it. Otherwise the `java.lang.NoClassDefFoundError` error can be thrown if anybody will use the `Instancio` tool with older version of validation library that does not contain these annotations.

Reply:
@nedis-ua, I forgot to mention that Hibernate validator 6 would be the minimum requirement.

Starting from version 6, all the constraints currently implemented are supported **except** `UUID`. We only need that special case for this single annotation.

https://docs.jboss.org/hibernate/validator/6.0/reference/en-US/html_single/#validator-defineconstraints-spec